src/order/complete_lattice.lean

```diff
@@ -1014,6 +1022,19 @@ end
 lemma infi_extend_top {e : ι → β} (he : injective e) (f : ι → α) : (⨅ j, extend e f ⊤ j) = infi f :=
 @supr_extend_bot αᵒᵈ _ _ _ _ he _
 
+lemma bsupr_const {ι : Sort*} {a : α} {s : set ι} (hs : s.nonempty) :
+  (⨆ (i ∈ s), a) = a :=
+begin
+  refine le_antisymm (supr₂_le (λ i hi, le_refl a)) _,
+  obtain ⟨i, hi⟩ := hs,
+  rw supr_split_single _ i,
+  simp [hi],
+end
```

Comment:
Use `supr_const` twice + `haveI`.

Reply:
I think this is not quite right. The inner `supr` produces a non-constant function and in general is not a `supr` over a non-empty type.

More precisely, consider:
```lean
import order.complete_lattice

variables {α : Type*} [complete_lattice α]

lemma bsupr_const {ι : Sort*} {a : α} {s : set ι} (hs : s.nonempty) :
  (⨆ (i ∈ s), a) = a :=
begin
  let f : ι → α := λ i, ⨆ (h : i ∈ s), a,
  change (⨆ i, f i) = a,
  classical,
  have hf : f = λ i, if i ∈ s then a else ⊥, { ext i, by_cases h : i ∈ s; simp [h, f], },

  sorry,
end
```

Reply:
Ah sorry, you can use `supr_subtype''` to fill the gap.

Reply:
Aha, this is the *right* proof, thanks!